### ISSUE TITLE
TypeError: list indices must be integers or slices, not str

### ISSUE THREAD

**Issue by Mu-Xiu-Yulin** [Significant-Gravitas/Auto-GPT#1483]
### Duplicates

- [X] I have searched the existing issues

### Steps to reproduce 🕹

```
Debug Mode:  ENABLED
Welcome back!  Would you like me to return to being Entrepreneur-GPT?
Continue with the last settings?
Name:  Entrepreneur-GPT
Role:  an AI designed to autonomously develop and run businesses with the sole goal of increasing your net worth.
Goals: ['Increase net worth', 'Grow Twitter Account', 'Develop and manage multiple businesses autonomously']
Continue (y/n): y
Using memory of type: LocalCache
  Token limit: 4000
  Memory Stats: (0, (0, 1536))
  Token limit: 4000
  Send Token Count: 890
  Tokens remaining for response: 3110
  ------------ CONTEXT SENT TO AI ---------------
  System: The current time and date is Sat Apr 15 13:40:45 2023

  System: This reminds you of these events from your past:




  User: Determine which next command to use, and respond using the format specified above:

  ----------- END OF CONTEXT ----------------
Creating chat completion with model gpt-3.5-turbo, temperature 1.0, max_tokens 3110
Traceback (most recent call last):
  File "E:\Miniconda\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "E:\Miniconda\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "D:\VScode\Auto-GPT\autogpt\__main__.py", line 572, in <module>
    main()
  File "D:\VScode\Auto-GPT\autogpt\__main__.py", line 396, in main
    agent.start_interaction_loop()
  File "D:\VScode\Auto-GPT\autogpt\__main__.py", line 448, in start_interaction_loop
    assistant_reply = chat.chat_with_ai(
  File "D:\VScode\Auto-GPT\autogpt\chat.py", line 159, in chat_with_ai
    assistant_reply = create_chat_completion(
  File "D:\VScode\Auto-GPT\autogpt\llm_utils.py", line 32, in create_chat_completion
    deployment_id=cfg.get_azure_deployment_id_for_model(model),
  File "D:\VScode\Auto-GPT\autogpt\config.py", line 111, in get_azure_deployment_id_for_model
    return self.azure_model_to_deployment_id_map["fast_llm_model_deployment_id"]
TypeError: list indices must be integers or slices, not str
```

### Your prompt 📝

```yaml
# Paste your prompt here
```

**Comment by shenqingwangyixuan** [comment]
我也遇到了这种问题，请问该如何解决。

**Comment by MSDNAndi** [comment]
same here

**Comment by Mu-Xiu-Yulin** [comment]
我要没有解决方案



------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2023年4月16日(星期天) 凌晨3:52
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; "State ***@***.***&gt;; 
主题: Re: [Significant-Gravitas/Auto-GPT] TypeError: list indices must be integers or slices, not str (Issue #1483)





 
我也遇到了这种问题，请问该如何解决。

**Comment by MSDNAndi** [comment]
Friends, I fixed it on my machine. I am running Windows. 
Problem is in autogpt/config/config.py
the line
    AZURE_CONFIG_FILE = os.path.join(os.path.dirname(__file__), "..", "azure.yaml")
leads to a folder like this
'...\\Auto-GPT\\autogpt\\config\\**..**\\azure.yaml'
the problem is that it will not go where the yaml file may be.
Before it was in the root directory of the repository.
so
Auto-GPT
and it gets called from
Auto-GPT/autogpt/config
so it looks in Auto-GPT/autogpt
so you need to put azure.yaml one level above config.py. Most people will have it 2 levels above.
Also I need to enter my values like this
chatgpt was my chosen name for Chat GPT Turbo, gpt432k - my name for the 32k token version of GPT4.

Compared to the examples I found online I changed azure-ad to just azure
I removed the trailing "/" from azure_api_base


azure_api_type: "azure"
azure_api_base: "https://MYDEPLOYMENTNAME.openai.azure.com"
azure_api_version: "2023-03-15-preview"
azure_model_map:
    fast_llm_model_deployment_id: "chatgpt"
    smart_llm_model_deployment_id: "gpt432k" 
    embedding_model_deployment_id: "textembeddingada002"

**Comment by Mu-Xiu-Yulin** [comment]
>

**Comment by Mu-Xiu-Yulin** [comment]
>

**Comment by Mu-Xiu-Yulin** [comment]
> 

No, my friend, maybe your method is a solution, but my solution is to change the USE_AZURE in the .env file to False, so that it can be used, I hope your solution can help others

**Comment by MSDNAndi** [comment]
if you change USE_AZURE to false, then you are not using Azure but OpenAI's API endpoints and you pay OpenAI directly, no Microsoft :) If that is what you need, that is of course OK.
This Azure configuration only makes sense when wanting to use Azure.
I cannot use OpenAI's own endpoint.

**Comment by Mu-Xiu-Yulin** [comment]
> Friends, I fixed it on my machine. I am running Windows. Problem is in autogpt/config/config.py the line AZURE_CONFIG_FILE = os.path.join(os.path.dirname(**file**), "..", "azure.yaml") leads to a folder like this '...\Auto-GPT\autogpt\config\**..**\azure.yaml' the problem is that it will not go where the yaml file may be. Before it was in the root directory of the repository. so Auto-GPT and it gets called from Auto-GPT/autogpt/config so it looks in Auto-GPT/autogpt so you need to put azure.yaml one level above config.py. Most people will have it 2 levels above. Also I need to enter my values like this chatgpt was my chosen name for Chat GPT Turbo, gpt432k - my name for the 32k token version of GPT4.
> 
> Compared to the examples I found online I changed azure-ad to just azure I removed the trailing "/" from azure_api_base
> 
> azure_api_type: "azure" azure_api_base: "https://MYDEPLOYMENTNAME.openai.azure.com" azure_api_version: "2023-03-15-preview" azure_model_map: fast_llm_model_deployment_id: "chatgpt" smart_llm_model_deployment_id: "gpt432k" embedding_model_deployment_id: "textembeddingada002"

No, my friend, maybe your method is a solution, but my solution is to change the USE_AZURE in the .env file to False, so that it can be used, I hope your solution can help others

**Comment by Mu-Xiu-Yulin** [comment]
> if you change USE_AZURE to false, then you are not using Azure but OpenAI's API endpoints and you pay OpenAI directly, no Microsoft :) If that is what you need, that is of course OK. This Azure configuration only makes sense when wanting to use Azure. I cannot use OpenAI's own endpoint.

Well, my friend, I hope your solution and mine can help others

**Comment by John-CYHui** [comment]
For others with the same issue, check this pull request https://github.com/Significant-Gravitas/Auto-GPT/pull/2214

**Comment by lc0rp** [comment]
Potentially fixed in #2355. Please retest. 
If the issue persists, please reopen or create a new bug.
